src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -627,6 +627,8 @@ internal bool IsNew
             get { return (_modifiers & DeclarationModifiers.New) != 0; }
         }
 
+        internal bool HasReadOnlyModifier => (_modifiers & DeclarationModifiers.ReadOnly) != 0;
```

Comment:
This might have to become public API at some point, so we'll have to look at naming. IsReadOnly means something else on properties.